### PR TITLE
Add the first part of the Exodii warehouse mission

### PR DESCRIPTION
**Pull request by SariusSkelrets**
#### Summary
Content "Add a new mission to the exodii, as described by #59905"

#### Purpose of change

The Exodii needs more missions and the warehouse mission is desired since last year

#### Describe the solution

Add the building, the dialog and the new item

#### Describe alternatives you've considered

none

#### Testing

Tested:

- [x] the warehouse spawned on the overmap and only once
- [x] the warehouse can be found before the mission if the player revealed a lot of map
- [x] Rubik can give the mission if they know about the Hub but not before
- [x] you gain the rebooter when starting and consume it to reboot the locator
- [x] the locator cannot be rebooted twice even if you somehow gain another rebooter
- [x] reaching the warehouse after getting the mission is sufficient to complete the mission. However, Rubik will comment on it and demand to be told the warehouse's exact location 
- [x] Rubik will comment if you talked with the Hub before accepting this mission
- [x] Rubik's dialog is completely implemented 

#### Additional context

Currently the warehouse won't be globally unique so the player have a possibility to find it. It will be made unique when NPCs will be able to give you directions to it

### CODE REVIEW

File: data/json/furniture_and_terrain/furniture-alien.json [hunk 1]
```diff
@@ -669,6 +669,55 @@
       ]
     }
   },
+  {
+    "type": "furniture",
+    "id": "f_exodii_locator_antenna",
+    "name": "locator antenna",
+    "description": "A tall metal pole solidly fixed to the roof. It does not look activated but there's a small set of holes near its base, maybe something could be plugged in one to reboot the locator.",
```
Comment (Procyonae) [inline data/json/furniture_and_terrain/furniture-alien.json:676]:
```suggestion
    "description": "A tall metal pole solidly fixed to the roof.  It does not look activated but there's a small set of holes near its base, maybe something could be plugged in one to reboot the locator.",
```

File: data/json/npcs/exodii/exodii_merchant_missions.json [hunk 2]
```diff
@@ -52,11 +52,35 @@
...
+      "success_lie": "Fine an' fine.  We can pass it in when you like.",
+      "failure": "Fine an' fine.  Come back to this'n if you like to try again."
+    }
+  },
   {
     "id": "TALK_EXODII_MERCHANT_mission_1_intro",
     "type": "talk_topic",
     "dynamic_line": "Them as offer as can, an it be so. Aye, there's summat the old Scythy been larkin' up Rubik's tailpipe.  B'ain't tass right for a clanker like this'n, but a pop-an-work like you'n, maybe well and suited.",
     "//~": "Just because you asked, I think we can work something out.  Yes, the Scythean has been bugging me about something.  It's not something a cyborg like me can do, but a mercenary would probably fit the bill.",
+    "run_eocs": { "id": "EOC_MISSION_EXODII_HUB_DRONE_CHECK" },
```
Comment (Procyonae) [inline data/json/npcs/exodii/exodii_merchant_missions.json:83]: It doesn't look like you can put "run_eocs" here it needs to be in "responses" or in "speaker_effect" looking at existing usage in talk_topic s (though I've no idea what the latter does)

File: data/json/npcs/exodii/exodii_merchant_missions.json [hunk 3]
```diff
@@ -242,5 +266,103 @@
...
+  },
+  {
+    "id": "TALK_EXODII_MERCHANT_mission_2_intro",
+    "type": "talk_topic",
+    "dynamic_line": {
+      "days_since_cataclysm": 28,
+      "yes": "Well an surely, fruit gum!",
+      "no": {
+        "compare_num": [ { "global_val": "var", "var_name": "u_brought_drone_at_exobase" }, "=", { "const": 1 } ],
+        "yes": "Well an surely, fruit gum!  Mayhap this be a chance to show y'self a true, for Rubik's got sommat to ask yet again. The way o' the Exode is t'shimmy an' wobble, an' on thislike wobble, a few bits wobbled off, kennit? 'Tis rightful common, alack. One such is hidden for our escolts an' our screeds. Mayhap as y'know the lands, ye'd have a see of it?",
```
Comment (Procyonae) [inline data/json/npcs/exodii/exodii_merchant_missions.json:278]:
```suggestion
        "yes": "Well an surely, fruit gum!  Mayhap this be a chance to show y'self a true, for Rubik's got sommat to ask yet again.  The way o' the Exode is t'shimmy an' wobble, an' on thislike wobble, a few bits wobbled off, kennit?  'Tis rightful common, alack.  One such is hidden for our escolts an' our screeds.  Mayhap as y'know the lands, ye'd have a see of it?",
```

File: data/json/npcs/exodii/exodii_merchant_missions.json [hunk 4]
```diff
@@ -242,5 +266,103 @@
...
+        "no": {
+          "compare_num": [ { "npc_val": "var", "var_name": "exodii_knows_hub01" }, "=", { "const": 1 } ],
+          "yes": "Well an surely, fruit gum! For an' such as ye've done us a good.",
```
Comment (Procyonae) [inline data/json/npcs/exodii/exodii_merchant_missions.json:281]:
```suggestion
          "yes": "Well an surely, fruit gum!  For an' such as ye've done us a good.",
```

File: data/json/npcs/exodii/exodii_merchant_missions.json [hunk 5]
```diff
@@ -242,5 +266,103 @@
...
+    },
+    "responses": [
+      { "text": "What's the mission?", "topic": "TALK_EXODII_MERCHANT_mission_2" },
+      { "text": "I've gotta go.  See you later, Rubik.", "topic": "TALK_DONE" }
+    ]
+  },
+  {
+    "id": "TALK_EXODII_MERCHANT_mission_2",
+    "type": "talk_topic",
+    "dynamic_line": "Aye, ye kennit well. 'Tis a store-cabin, mayhap the likes you've seen. This'n be older'n most, clade o' iron and crete-stone. Abreast the ceil, high atop, there be a steely jut, piercin' the heavens. In times ordinary, Great Grey could call out and screed the jut, kennit, and send aloft an escolt to razz 'er. Oft times, the jut is wibbled on the wobble, an' it be coals'n coke. Ol' Rubik can give you'n a dandy to pop 'er back aloft, and that's a fig and a fiddle.",
```
Comment (Procyonae) [inline data/json/npcs/exodii/exodii_merchant_missions.json:294]:
```suggestion
    "dynamic_line": "Aye, ye kennit well.  'Tis a store-cabin, mayhap the likes you've seen.  This'n be older'n most, clade o' iron and crete-stone.  Abreast the ceil, high atop, there be a steely jut, piercin' the heavens.  In times ordinary, Great Grey could call out and screed the jut, kennit, and send aloft an escolt to razz 'er.  Oft times, the jut is wibbled on the wobble, an' it be coals'n coke.  Ol' Rubik can give you'n a dandy to pop 'er back aloft, and that's a fig and a fiddle.",
```

**Review by Procyonae — CHANGES_REQUESTED** [review]

File: data/json/npcs/exodii/exodii_merchant_missions.json (hunk 2 above)
Comment (SariusSkelrets) [reply data/json/npcs/exodii/exodii_merchant_missions.json:83]: I'll try to see if there's a way to run that EoC every time you talk to Rubik without putting it everywhere. 

Checking for interactions with Rubik is a sure way to ensure both your presence and you being correlated to the drone's arrivale

Comment (SariusSkelrets) [reply data/json/npcs/exodii/exodii_merchant_missions.json]: The logic was reworked as suggested on discord. It now checks if you ever interacted with hub instead

**Comment by SariusSkelrets** [comment]
@I-am-Erk the mission is now functionnal. I would greatly appreciate if you could help me with Rubik's lines

**Comment by I-am-Erk** [comment]
I've added a bit to the GitHub gist. I'm hoping to pop in and just directly code it in for you this evening depending on the next couple hours. My kids are out of town so it's looking hopeful

**Comment by I-am-Erk** [comment]
The CBM choices here don't have the be the final ones. I will take a look at the remaining dialogue too and try to expand it a bit, but it's my bedtimr

File: data/json/npcs/exodii/exodii_merchant_talk.json [hunk 6]
```diff
@@ -735,7 +744,13 @@
           "no": [
             {
               "compare_int": [ { "global_val": "var", "var_name": "exodii_knows_hub01" }, "==", { "const": 1 } ],
-              "yes": ", but for the right-an-quick, this'n has nothin' t'be tassed for ye.",
+              "yes": [
+                {
+                  "compare_int": [ { "global_val": "var", "var_name": "warehouse_was_found" }, "==", { "const": 1 } ],
+                  "yes": ", but for the right-an-quick, this'n has nothin' t'be tassed for ye.",
+                  "no": ".  How'd ye be twigged by a (((SWAP_FOR_EXOSPEAK_treasure hunt for a lost warehouse_SWAP_FOR_EXOSPEAK)))?"
```
Comment (I-am-Erk) [inline data/json/npcs/exodii/exodii_merchant_talk.json:751]:
```suggestion
                  "no": ".  How'd ye be twigged by a pop-an-squeak?  Or as y'r tassed, a treasure hunt?"
```

File: data/json/npcs/exodii/exodii_merchant_missions.json [hunk 7]
```diff
@@ -242,5 +265,199 @@
...
+      { "text": "Any clues on how to find it?", "topic": "TALK_EXODII_MERCHANT_mission_2_find" },
+      { "text": "What's in this for me?", "topic": "TALK_EXODII_MERCHANT_mission_2_reward" },
+      { "text": "I need a moment.  Can we talk about something else?", "topic": "TALK_EXODII_MERCHANT_Talk" },
+      { "text": "I'm gonna need to think on this one.  See you later, Rubik.", "topic": "TALK_DONE" }
+    ]
+  },
+  {
+    "id": "TALK_EXODII_MERCHANT_mission_2_find",
+    "type": "talk_topic",
+    "dynamic_line": "That's be for your assert, an I'm ken. If ol' Rubik could find 'er, us'n'd send out the stompers. For a yark an' a lark, though, couldst ask another of your sameworld?  Mercants or traveler-such?  It be a fine slick to a local guide.",
```
Comment (Maleclypse) [inline data/json/npcs/exodii/exodii_merchant_missions.json:379]:
```suggestion
    "dynamic_line": "That's be for your assert, an I'm ken. If ol' Rubik could find 'er, us'n'd send out the stompers.  For a yark an' a lark, though, couldst ask another of your sameworld?  Mercants or traveler-such?  It be a fine slick to a local guide.",
```

File: data/json/npcs/exodii/exodii_merchant_missions.json [hunk 8]
```diff
@@ -242,5 +265,199 @@
...
+      { "text": "What's in this for me?", "topic": "TALK_EXODII_MERCHANT_mission_2_reward" },
+      {
+        "text": "Then I'll find it.",
+        "topic": "TALK_DONE",
+        "effect": { "assign_mission": "EXODII_MISSION_WAREHOUSE" }
+      },
+      {
+        "text": "Beloved jaryi?  What did you just call me?",
+        "topic": "TALK_EXODII_MERCHANT_jaryi",
+        "arithmetic": [ { "global_val": "var", "var_name": "u_know_jaryi" }, "=", { "const": 1 } ]
```
Comment (Maleclypse) [inline data/json/npcs/exodii/exodii_merchant_missions.json:331]: Not sure what this is supposed to be.

Comment (I-am-Erk) [reply data/json/npcs/exodii/exodii_merchant_missions.json:331]: Variable tracking that can be used to add more conversation topics about where Rubik is from. I'm not a huge fan of using arithmetic for that stuff myself

Comment (Maleclypse) [reply data/json/npcs/exodii/exodii_merchant_missions.json:331]: Sorry I wasn't clear in my comment there's an error there saying that arithmetic can't be used that way.

Comment (SariusSkelrets) [reply data/json/npcs/exodii/exodii_merchant_missions.json]: I fixed that line. Thanks for pointing it to me

**Pull request merged**